build/config/build.yaml

```diff
@@ -82,4 +87,61 @@ steps:
   dir: 'tests/'
   waitFor: ['DOCKER_STAGE', 'DOCKER_BUILD_MVN_GCLOUD']
 
+#
+# End of App Engine Integration tests
+
+#
+# GKE Integration tests
+#
+
+# Create a Docker image with the test application
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'DOCKER_BUILD_TEST_APPLICATION'
+  args: ['build', '--tag=${_STAGING_IMAGE}-test-app', '--no-cache', 'tests/runtimes-common-testing/target/deploy']
```

Comment:
I think it's possible that the files in the `tests/runtimes-common-testing/target` directory will not be present when this step runs because the JETTY_SMOKE_TEST runs `mvn clean install`. @meltsufin ran into a similar issue as well. 

I think maybe the best solution is to modify our maven executions so the maven project is only built once?